txpool/txpool.go

```diff
@@ -347,6 +354,11 @@ func (p *TxPool) getSealing() bool {
 	return atomic.LoadUint32(&p.sealing) == 1
 }
 
+// SetBaseFee sets the given base fee
+func (p *TxPool) SetBaseFee(baseFee uint64) {
```

Comment:
Could we reuse `Prepare` function (i.e. Prepare(baseFee uint)? `baseFee` is a parameter of the block and `Prepare` is called before building a block.

Reply:
Moved the passing base fee to tx pool in a separate PR to review it easily - https://github.com/0xPolygon/polygon-edge/pull/1243. And applied your suggestion.